lib/multi.c

```diff
@@ -1107,12 +1131,33 @@ static CURLMcode Curl_multi_wait(struct Curl_multi *multi,
   curlfds = nfds; /* number of internal file descriptors */
   nfds += extra_nfds; /* add the externally provided ones */
 
+#ifdef USE_WINSOCK
+  if(extra_nfds > 0) {
+    uevents[nevents] = WSACreateEvent();
```

Comment:
Can't you use the same event for normal data fds and extra_fds? Why is it better to have them separated?

Reply:
Actually I think the events for sockets cannot be re-used at all with the current logic, because this will change the meaning of the pollret return code and limit it to only 1 or 2 instead of the actual number of ready socket handles. Or am I missing something?

>        On success, poll() returns a nonnegative value which is the number of
>        elements in the pollfds whose revents fields have been set to a
>        nonzero value (indicating an event or an error).  A return value of
>        zero indicates that the system call timed out before any file
>        descriptors became read.
> 
>        On error, -1 is returned, and errno is set to indicate the cause of
>        the error.

Reply:
Ah! You're right. This can be solved by counting all the sockets that were actually listed here. I'll update the PR to do this, and also switch to allocating a single event when creating the multi handle, and reusing it for all sockets and for explicit wakes.